python/Ganga/Core/GangaRepository/Registry.py

```diff
@@ -883,6 +931,8 @@ def startup(self):
             self.repository.startup()
             # All Ids could have changed
             self.changed_ids = {}
+            self.flush_thread = RegistryFlusher(self)
+            self.flush_thread.start()
```

Comment:
Which is why you would write a test with and without the background thread manipulating things on you whilst you're working or trying to work.